### ISSUE TITLE
code snipets: add Reset and Copy buttons

### ISSUE THREAD

**Issue by lypwig** [tidalcycles/strudel#576]
I'm reading the documentation (wow very nice work here) and I have two suggestions for the code snippets:

- a button to reset the content to the initial state. Useful when the user modified the code to a broken state and want to make it work again;
- a button to copy the content in the clipboard, like in the GitHub UI (on the right):

```
foo bar
```

I can try to work on it if you like these proposals.

**Comment by felixroos** [comment]
good idea! maybe it makes sense to still make them optional via the component props. Also worth noting that the error messages are on the top right, so they need some space there

**Comment by roipoussiere** [comment]
> Also worth noting that the error messages are on the top right, so they need some space there

Good point! What about having the error messages on the left and the new buttons on the right like this (I will change button icons of course but you get the idea)?

![image](https://github.com/tidalcycles/strudel/assets/1665542/8fb0b34f-9619-4a65-a019-a36a3e52657a)

**Comment by felixroos** [comment]
> Good point! What about having the error messages on the left and the new buttons on the right like this (I will change button icons of course but you get the idea)?

Could work, but I wonder what happens if the error message is multiline? Another idea would be to move the error message to the bottom, extending the height of the container if there is one.

I am also preparing a workshop where the mini repl has only the height of the editor, which is useful as inline content, either inline in a text or in a table like this:

<img width="780" alt="image" src="https://github.com/tidalcycles/strudel/assets/12023032/b85cf550-170d-483f-8b18-6f8947d6d0fb">

There are currently no buttons yet (only keyboard controls), but I plan to add them on the right of the same line.

**Comment by roipoussiere** [comment]
> I wonder what happens if the error message is multiline?

Oh I didn't know this could happen. Ok for moving it to the bottom.

> I am also preparing a workshop where the mini repl has only the height of the editor

nice!

**Comment by cleary** [comment]
> a button to reset the content to the initial state. Useful when the user modified the code to a broken state and want to make it work again;

I came here to suggest the same ... +1 :)
